src/internal/packager2/publish.go

```diff
@@ -0,0 +1,208 @@
+		PublicKeyPath:           opts.PublicKeyPath,
+		SkipSignatureValidation: opts.SkipSignatureValidation,
+	}
+	pkgLayout, err := layout2.LoadFromTar(ctx, path, layoutOpts)
+	if err != nil {
+		return fmt.Errorf("unable to load package: %w", err)
+	}
+
+	return pushToRemote(ctx, pkgLayout, dst, opts.Concurrency, opts.WithPlainHTTP)
+}
+
+// PublishSkeletonOpts declares the parameters to publish a skeleton package.
+type PublishSkeletonOpts struct {
+	// Concurrency configures the zoci push concurrency if empty defaults to 3.
+	Concurrency int
+	// SigningKeyPath points to a signing key on the local disk.
+	SigningKeyPath string
+	// SigningKeyPassword holds a password to use the key at SigningKeyPath.
+	SigningKeyPassword string
+	// WithPlainHTTP falls back to plain HTTP for the registry calls instead of TLS.
+	WithPlainHTTP bool
+}
+
+// PublishSkeleton takes a Path to the location of the build package, a ref to a registry, and a PublishOpts and uploads the
+// package tarball, oci reference, or skeleton package to the registry.
+func PublishSkeleton(ctx context.Context, path string, ref registry.Reference, opts PublishSkeletonOpts) error {
+	l := logger.From(ctx)
+
+	// Validate inputs
+	l.Debug("validating PublishOpts")
+	if err := ref.ValidateRegistry(); err != nil {
+		return fmt.Errorf("invalid registry: %w", err)
+	}
+	if path == "" {
+		return fmt.Errorf("path must be specified")
+	}
+
+	// Load package layout
+	l.Info("loading skeleton package", "path", path)
+	// Create skeleton buildpath
+	createOpts := layout2.CreateOptions{
+		SigningKeyPath:     opts.SigningKeyPath,
+		SigningKeyPassword: opts.SigningKeyPassword,
+		SetVariables:       map[string]string{},
+	}
+	buildPath, err := layout2.CreateSkeleton(ctx, path, createOpts)
+	if err != nil {
+		return fmt.Errorf("unable to create skeleton: %w", err)
+	}
+
+	layoutOpts := layout2.PackageLayoutOptions{
+		SkipSignatureValidation: true,
+		IsPartial:               false,
+	}
+	pkgLayout, err := layout2.LoadFromDir(ctx, buildPath, layoutOpts)
+	if err != nil {
+		return fmt.Errorf("unable to load skeleton: %w", err)
+	}
+
+	return pushToRemote(ctx, pkgLayout, ref, opts.Concurrency, opts.WithPlainHTTP)
+}
+
+// pushToRemote pushes a package to a remote at ref.
+func pushToRemote(ctx context.Context, layout *layout2.PackageLayout, ref registry.Reference, concurrency int, plainHTTP bool) error {
+	// Build Reference for remote from registry location and pkg
+	r, err := layout2.ReferenceFromMetadata(ref.String(), layout.Pkg)
+	if err != nil {
+		return err
+	}
+
+	arch := layout.Pkg.Metadata.Architecture
+	// Set platform
+	p := oci.PlatformForArch(arch)
+
+	// Set up remote repo client
+	rem, err := layout2.NewRemote(ctx, r, p, oci.WithPlainHTTP(plainHTTP))
+	if err != nil {
+		return fmt.Errorf("could not instantiate remote: %w", err)
+	}
+
+	logger.From(ctx).Info("pushing package to remote registry",
+		"ref", ref,
+		"architecture", arch)
+	return rem.Push(ctx, layout, concurrency)
```

Comment:
I would recommend deleting this log statement as `rem.Push` immediately does a similar log statement so the UX feels a bit awkward
![image](https://github.com/user-attachments/assets/f9e20fc3-2b0f-4287-8b81-bbd39048d3fe)
I'd be in favor of adding architecture to the `rem.Push` log with `layout.Pkg.Metadata.Architecture` though.

Reply:
Also in favor of this change